.github/actions/cache-query-compilation/action.yml

```diff
@@ -0,0 +1,60 @@
+name: Cache query compilation
+description: Caches CodeQL compilation caches - should be run both on PRs and pushes to main.
+
+inputs:
+  key:
+    description: 'The cache key to use - should be unique to the workflow'
+    required: true
+
+outputs:
+  cache-dir:
+    description: "The directory where the cache was stored"
+    value: ${{ steps.fill-compilation-dir.outputs.compdir }}
+
+runs:
+  using: composite
+  steps:
+    # Cache the query compilation caches.
+    # calculate the merge-base with main, in a way that works both on PRs and pushes to main. 
+    - name: Calculate merge-base
+      shell: bash
+      if: ${{ github.event_name == 'pull_request' }}
+      env:
+        BASE_BRANCH: ${{ github.base_ref }}
+      run: |
+        MERGE_BASE=$(git cat-file commit $GITHUB_SHA | grep '^parent ' | head -1 | cut -f 2 -d " ")
+        echo "merge-base=$MERGE_BASE" >> $GITHUB_ENV
```

Comment:
`merge-base` is not a valid environment variable in `bash`. You cant write `${merge-base}` in a script. I'd recommend using `merge_base`  or `MERGE_BASE`.

Reply:
Is there a reason for using `GITHUB_ENV` here instead of `GITHUB_OUTPUT` ?

Reply:
> Is there a reason for using GITHUB_ENV here instead of GITHUB_OUTPUT ?

Nope, it was just the first data-sharing mechanism I found.  
But now that I look at it, I still like it. I don't have to do `${{ steps.step-name.outputs.output-name }}`, but can just do `${{ env.merge_base }}`.

Reply:
> merge-base is not a valid environment variable in bash. You cant write ${merge-base} in a script. I'd recommend using merge_base or MERGE_BASE.

It works with `merge-base` 🤷  
I changed it anyway.

Reply:
Thanks! It's true that `merge-base` works, however, things get annoying if you want to print the value in a bash script for debugging reasons. It's just to prevent our future selves from scratching our heads why `${merge-base} == "base"` ;-)